src/report.ts

```diff
@@ -93,10 +100,19 @@ export class ReportModule {
         return (error as ApiResponseError).response !== undefined;
     }
 
-    private convertFilter(filter: AFM.CompatibilityFilter): AFM.CompatibilityFilter | null {
-        if ((filter as AFM.IExpressionFilter).value === undefined) {
-            return convertVisualizationObjectFilter(filter as VisualizationObjectFilter);
+    private sanitizeFilters(filter: AFM.CompatibilityFilter): ExecuteAFM.CompatibilityFilter | null {
+        if (AFM.isAttributeFilter(filter)) {
+            return convertAttributeFilter(filter);
+        } else if (AFM.isAbsoluteDateFilter(filter)) {
+            return convertAbsoluteDateFilter(
+                filter as VisualizationObject.IVisualizationObjectAbsoluteDateFilter,
```

Comment:
we don't like these explicit changes of type. If these two structures are completely identical, only have two different names (types) than converter implementation should accept both of them as input param ->
`function convertRelativeDateFilter(
    filter: IVisualizationObjectRelativeDateFilter | AFM.CompatibilityFilter,
): AFM.FilterItem | null {`

Reply:
updated. thank you :)